File: original artifact/Ian_Foss_cs-360_project_three_inventory_app/cs360_project_three/app/src/main/java/com/example/cs_360_project_three/InventoryItemEntity.java
package com.example.cs_360_project_three;

import androidx.room.Entity;
import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;

import androidx.room.PrimaryKey;


// Entity class for the inventory items
@Entity(tableName = "inventory_table")
public class InventoryItemEntity {


    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    private long id;

    @NonNull
    @ColumnInfo(name = "name")
    private String name;

    @NonNull
    @ColumnInfo(name = "quantity")
    private int quantity;

    public InventoryItemEntity(String name, int quantity) {
        this.name = name;
        this.quantity = quantity;
    }

    public @NonNull String getName() {
        return name;
    }


    public @NonNull void setName(String name) {
        this.name = name;
    }

    public  @NonNull int getQuantity() {
        return quantity;
    }

    public @NonNull void setQuantity(int quantity) {
        this.quantity = quantity;
    }


    public long getId() {
        return id;
    }


    public void setId(long id) {
        this.id = id;
    }


}

File: enhancements/Enhancement Two/app/src/main/java/com/example/CS499_Capstone/InventoryItemEntity.kt
package com.example.CS499_Capstone

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

// Entity class for the inventory items
@Entity(tableName = "inventory_table")
data class InventoryItemEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long = 0,

    @ColumnInfo(name = "name")
    var name: String,

    @ColumnInfo(name = "quantity")
    var quantity: Int

)
